### ISSUE TITLE
python api

### ISSUE THREAD

**Issue by cestas** [aaalgo/kgraph#17]
hi 
i am trying to use the python API but cannot seem to make the pykgraph package.
I am using a linux16.04 machine.
i cloned the repository onto my local computer in a folder and ran the initial compilation steps:
cmake -DCMAKE_BUILD_TYPE=release .
make
sudo make install
These went fine. when i run python setup.py install the only file that is created is pykgraph-2.0.egg-info in a folder where I have other python packages. There is no package folder 'pykgraph' that i can import into a python script which would then make a KNN graph. (I am using python 3.5)
when i try to import pykgraph i get a segmentation default
shobi@shobi:~/Documents/kgraph$ python
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pykgraph
Segmentation fault (core dumped)
shobi@shobi:~/Documents/kgraph$ 

Also, The first time i run setup.py install i get the following warnings, but no errors:
running install
running build
running build_ext
building 'pykgraph' extension
creating build
creating build/temp.linux-x86_64-3.5
creating build/temp.linux-x86_64-3.5/python
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I. -I/home/shobi/.local/lib/python3.5/site-packages/numpy/core/include -I/usr/include/python3.5m -c kgraph.cpp -o build/temp.linux-x86_64-3.5/kgraph.o -O3 -std=c++11 -msse2 -fopenmp -DKGRAPH_VERSION=b'2372db6\n'
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I. -I/home/shobi/.local/lib/python3.5/site-packages/numpy/core/include -I/usr/include/python3.5m -c metric.cpp -o build/temp.linux-x86_64-3.5/metric.o -O3 -std=c++11 -msse2 -fopenmp -DKGRAPH_VERSION=b'2372db6\n'
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I. -I/home/shobi/.local/lib/python3.5/site-packages/numpy/core/include -I/usr/include/python3.5m -c python/pykgraph.cpp -o build/temp.linux-x86_64-3.5/python/pykgraph.o -O3 -std=c++11 -msse2 -fopenmp -DKGRAPH_VERSION=b'2372db6\n'
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
In file included from /home/shobi/.local/lib/python3.5/site-packages/numpy/core/include/numpy/ndarraytypes.h:1816:0,
                 from /home/shobi/.local/lib/python3.5/site-packages/numpy/core/include/numpy/ndarrayobject.h:18,
                 from python/pykgraph.cpp:5:
/home/shobi/.local/lib/python3.5/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by " \
  ^
python/pykgraph.cpp: In function ‘PyObject* {anonymous}::load_lshkit(const string&, PyObject*)’:
python/pykgraph.cpp:64:29: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if (!(descr->elsize == elsize)) throw kgraph::io_error(path);
                             ^
python/pykgraph.cpp:66:44: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         if ((!nd) || (PyArray_ITEMSIZE(nd) != elsize)) throw kgraph::runtime_er
                                            ^
python/pykgraph.cpp: In function ‘void* init()’:
python/pykgraph.cpp:459:26: warning: control reaches end of non-void function [-Wreturn-type]
 init() { import_array(); }
                          ^
creating build/lib.linux-x86_64-3.5
x86_64-linux-gnu-g++ -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.5/kgraph.o build/temp.linux-x86_64-3.5/metric.o build/temp.linux-x86_64-3.5/python/pykgraph.o -lboost_python -lboost_timer -o build/lib.linux-x86_64-3.5/pykgraph.cpython-35m-x86_64-linux-gnu.so -fopenmp
running install_lib
copying build/lib.linux-x86_64-3.5/pykgraph.cpython-35m-x86_64-linux-gnu.so -> /usr/local/lib/python3.5/dist-packages
running install_egg_info
Removing /usr/local/lib/python3.5/dist-packages/pykgraph-2.0.egg-info
Writing /usr/local/lib/python3.5/dist-packages/pykgraph-2.0.egg-info
Thank you for your help
Shobi

**Comment by aaalgo** [comment]
I can confirm this with python3 and am working on it.  Python2 is OK.

**Comment by aaalgo** [comment]
I've fixed my test case.  I was linking against python2 version of boost_python.  The error went away after I made sure I was linking against the python3 version.  So I'm closing this issue.

After you run "python setup.py build", use "ldd build/lib*/pykgrah*.so" to find out the exact boost_python .so file you are using.  Python 3 version is usually called "libboost_python3.so....".  In my case, I have to modify setup.py and change boost_python to boost_python3 to make it work.
